src/NuGet.Core/NuGet.DependencyResolver.Core/LibraryRangeCacheKey.cs

```diff
@@ -10,9 +10,9 @@
 namespace NuGet.DependencyResolver
 {
     /// <summary>
-    /// Helper class to hold a library range and framework.
+    /// Helper type to hold a library range and framework.
     /// </summary>
-    public class LibraryRangeCacheKey : IEquatable<LibraryRangeCacheKey>
+    public readonly struct LibraryRangeCacheKey : IEquatable<LibraryRangeCacheKey>
```

Comment:
As mentioned in the PR description, this is a breaking change. I searched the type on GitHub, and all I can find are forks of this repo, I can't find any external usage of the type. Similarly, internally I couldn't find any usage. Finally, given my limited knowledge of NuGet.DependencyResolver.Core, I don't expect this to be a library that many people will use, and this type might not be needed by most users of the resolver. There's only 1 reference to the type outside of the assembly, which is a class used for caching, in `NuGet.Commands`.

FWIW, Prism's Allocation blame put this as NuGet's 6th highest cause of allocations in VS 17.6. 

So normally I'd be fairly conservative about breaking API changes, but this is a top cause of allocations and it's just really hard to imagine anyone actually using this API. I'm leaning towards yes, we should approve it. But if others have feedback, I'd like to hear it.

I'm really surprised that the public API analyzer didn't catch this as a breaking change. I wonder if we need to update the analyzer package version.

Reply:
I can't find any usages of the type either.  I lean towards making the change as the risk seems low.

Reply:
Is this type used outside of NuGet.DependencyResolver.Core? 
Like, is there a chance to make it internal altogether.

Reply:
> is there a chance to make it internal altogether

I'll try that and push an update.

Looking at a different NuGet GCPauseWatson issue with a teammate, again the challenge with fixing it was modifying public API. If there's an opportunity to make a whole bunch of NuGet code internal in one hit, that could give a lot more freedom to improve code quality, and avoid having to pay the risk analysis/monitoring every time you consider such a change. I don't have enough context to know whether that is possible though.

Reply:
Making `LibraryRangeCacheKey` internal would require making `ResolverUtility.FindLibraryCachedAsync` internal too and giving IVT to `NuGet.Commands` (or maybe linking the source into both projects?).

Reply:
@zivkan I'm guessing you're ok with this being merged?